internal/promotion/promotion.go

```diff
@@ -193,7 +208,7 @@ func (s *Step) Skip(
 	state promotion.State,
 ) (bool, error) {
 	if s.If == "" {
-		return false, nil
+		s.If = "${{ success() }}"
```

Comment:
1. Was this added because the meaning of `success()` may change at some point, and we want to continue to have consistent behavior?
2. As this takes a pointer to `s`, do we _actually_ want to modify it?
3. If "yes" on 1), I think this defaulting behavior requires documentation and reasoning.

Reply:
The default behavior is intended to be that if no expression is given, it's equivalent to `${{ success() }}`. Whether I took myself too literally or just got lazy and/or saw an opportunity for an easy implementation, perhaps we'll never know.

But I don't think it would be too hard to do this differently and it would be a bit efficient to avoid evaluating this "extra" expression. I'll change it.